src/Storage/AuthorizationCodeStorage.php

```diff
@@ -5,120 +5,112 @@
+	const META_KEY_PREFIX = 'oidc';
 
 	private static array $authorization_code_data = array(
 		'code'         => 'string', // authorization code.
 		'client_id'    => 'string', // client identifier.
-		'user_login'   => 'string', // The WordPress user id.
 		'redirect_uri' => 'string', // redirect URI.
 		'expires'      => 'int',    // expires as unix timestamp.
 		'scope'        => 'string', // scope as space-separated string.
 		'id_token'     => 'string', // The OpenID Connect id_token.
 	);
 
-	public function __construct() {
-		// Store the authorization codes in a taxonomy.
-		register_taxonomy( self::TAXONOMY, null );
-		foreach ( self::$authorization_code_data as $key => $type ) {
-			register_term_meta(
-				self::TAXONOMY,
-				$key,
-				array(
-					'type'              => $type,
-					'single'            => true,
-					'sanitize_callback' => array( __CLASS__, 'sanitize_' . $key ),
-				)
-			);
-		}
-	}
+	private function getCodeMeta( $code ) {
+		global $wpdb;
 
-	public static function sanitize_string_length( $string, $length ) {
-		return substr( $string, 0, $length );
-	}
+		if ( empty( $code ) ) {
+			return null;
+		}
 
-	public static function sanitize_code( $code ) {
-		return self::sanitize_string_length( $code, 40 );
-	}
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
+		$meta_row = $wpdb->get_row(
+			$wpdb->prepare(
+				"SELECT user_id, meta_key FROM $wpdb->usermeta WHERE meta_key LIKE %s;",
+				'%' . $wpdb->esc_like( $code ),
+			)
+		);
 
-	public static function sanitize_client_id( $client_id ) {
-		return self::sanitize_string_length( $client_id, 200 );
-	}
+		if ( empty( $meta_row ) ) {
+			return null;
+		}
 
-	public static function sanitize_redirect_uri( $redirect_uri ) {
-		return self::sanitize_string_length( $redirect_uri, 2000 );
+		return array(
+			'user_id'   => absint( $meta_row->user_id ),
+			'client_id' => str_replace( 'oidc_', '', str_replace( '_code_' . $code, '', $meta_row->meta_key ) ),
+		);
 	}
 
-	public static function sanitize_scope( $scope ) {
-		return self::sanitize_string_length( $scope, 100 );
-	}
+	public function getAuthorizationCode( $code ) {
+		$meta = $this->getCodeMeta( $code );
+		if ( empty( $meta ) ) {
+			return null;
+		}
 
-	public static function sanitize_id_token( $id_token ) {
-		return self::sanitize_string_length( $id_token, 2000 );
-	}
+		$user_id   = $meta['user_id'];
+		$client_id = $meta['client_id'];
+
+		$authorization_code = array(
+			'user_id'   => $user_id,
+			'client_id' => $client_id,
+			'code'      => $code,
+		);
+		foreach ( array_keys( self::$authorization_code_data ) as $key ) {
+			if ( 'code' === $key || 'client_id' === $key ) {
+				continue;
+			}
 
-	public static function sanitize_user_login( $user_login ) {
-		return self::sanitize_string_length( $user_login, 60 );
-	}
+			$meta_key                   = self::META_KEY_PREFIX . '_' . $client_id . '_' . $key;
```

Comment:
Yes! But since you mentioned namespacing the data per client id, I made it part of the key, because of which I had to use the `%LIKE` but if that triggers a full scan irrespective of index, I will think of another way of isolating the data on client id level, without making it part of the key.